src/cowboy_http2.erl

```diff
@@ -406,6 +428,10 @@ data_frame(State0=#state{opts=Opts, flow=Flow, streams=Streams}, StreamID, IsFin
 			State0
 	end.
 
+headers_frame(State=#state{http2_status=closing}, StreamID, _, _, _, _) ->
+	%% Don't accept a new stream after sending GOAWAY with a last stream ID.
+	reset_stream(State, StreamID, {stream_error, refused_stream,
+		'New streams not accepted while going away. (RFC7540 8.1.4)'});
```

Comment:
I don't think you need to send the error since we already sent a GOAWAY. On the other hand it must ignore streams larger than the stream_id we sent in the GOAWAY frame, including *all* frames, not just HEADERS.

Reply:
Good point. It only need to be ignored for HEADERS though, because it's the only frame from the client which can create new streams. I see now that headers_frame/6 is used for push also, so I'm moving the logic to frame/2.

Reply:
No you need to ignore all frames because a HEADERS frame for a new stream can be immediately followed with DATA frames for example. There's no round trip before DATA frames can be sent.

Reply:
DATA and RST_STREAM frames are already ignored if their StreamID is not a key in the map `State#state.streams`. HEADERS is the only frame for which we create a new key in the state streams map. Therefore, I don't think there is any other case which needs to be handled.

Reply:
OK. What about WINDOW_UPDATE? Also note that there might be things in the `cow_http2_machine` code.

Reply:
You're right of course. `cow_http2_machine` produces an error for frames referring no non-existent streams, which are forwarded back to the client.

I added a step before frame/2. (I also added some code which probably should belong in cowlib.)